docs/src/v1-api/problems.md

```diff
@@ -0,0 +1,22 @@
+# Problems of the current dream2nix
+
+## Integration of existing lang2nix tools
+Until now, integrating existing 2nix solutions into dream2nix was hard because dream2nix imposed standards which are not met by most existing tools.
+With v1 we want to lift most of these restrictions to make integration a no-brainer
+
+(see sections [integrate lang2nix tool (pure)](../v1-api/integrate-lang2nix-pure.md) and [integrate lang2ix tool (impure)](../v1-api/integrate-lang2nix-impure.md).
+
+## Tied to flakes
+The current api is tied to flakes. The v1 API should not depend on flakes anymore.
+
+## Composability
```

Comment:
This is the rationale in encouraging the use of blueprints/package-functions/proto-derivations (naming in-progress). The goal would be to make flakes easier to compose, or to compose in different ways.